src/connectivity/components.c

```diff
@@ -400,21 +400,21 @@ int igraph_is_connected(const igraph_t *graph, igraph_bool_t *res,
 
 int igraph_is_connected_weak(const igraph_t *graph, igraph_bool_t *res) {
 
-    long int no_of_nodes = igraph_vcount(graph);
+    long int added_count, no_of_nodes = igraph_vcount(graph);
     char *already_added;
     igraph_vector_t neis = IGRAPH_VECTOR_NULL;
     igraph_dqueue_t q = IGRAPH_DQUEUE_NULL;
 
-    long int i, j;
-
+    /* By convention, the null graph is not considered connected.
+     * See https://github.com/igraph/igraph/issues/1538 */
     if (no_of_nodes == 0) {
-        *res = 1;
+        *res = 0;
```

Comment:
Yes, I will add one for `igraph_is_connected()`.

Reply:
Note that this was not actually broken because `igraph_is_connected_weak` is just an internal function only ever called from `igraph_is_connected`. This "fix" was just for the sake of consistency. I added tests for `igraph_is_connected` though.